### PR TITLE
chore(update): update modal added

### PR DESCRIPTION
**Pull request by WanderingHogan**
### What this PR does 📖
Changes the 'update available' button to have a modal with directions so users don't get confused

Screenshot 2023-04-28 at 3 47 54 PM
Screenshot 2023-04-28 at 3 48 25 PM

- 

### Which issue(s) this PR fixes 🔨

- Resolve #

### CODE REVIEW

File: common/locales/en-US/main.ftl [hunk 1]
```diff
@@ -35,6 +35,14 @@ uplink = Uplink
     .check-for-updates = Check for updates
     .download-update = Download Update
 
+updates = Updates
+    .title = Uplink has an update available. Follow these steps to download and install the latest update.
+    .instruction1 = 1. Download the update.
+    .instruction2 = 2. Close uplink.
```
Comment (stavares843) [inline common/locales/en-US/main.ftl]:
```suggestion
    .instruction2 = 2. Close Uplink.
```

File: common/locales/en-US/main.ftl [hunk 2]
```diff
@@ -35,6 +35,14 @@ uplink = Uplink
...
+    .instruction3 = 3. Run the installer. The older version of uplink will be overwritten, leaving your account intact.
+    .instruction4 = 4. Open uplink and start chatting again!
```
Comment (stavares843) [inline common/locales/en-US/main.ftl]:
```suggestion
    .instruction4 = 4. Open Uplink and start chatting again!
```

**Review by sdwoodbury — APPROVED** [review]

File: common/locales/en-US/main.ftl [hunk 3]
```diff
@@ -35,6 +35,15 @@ uplink = Uplink
...
 
+updates = Updates
+    .title = Uplink has an update available. Follow these steps to download and install the latest update.
+    .instruction1 = 1. Download the update from our release page.
+    .instruction2 = 2. Close Uplink.
+    .instruction3 = 3. Run the installer. The older version of uplink will be overwritten, leaving your account intact.
+    .instruction4 = 4. Open Uplink and start chatting again!
+    .button-label = Pick Download Folder
+    .download-label = Download Update
+
```
Comment (stavares843) [inline common/locales/en-US/main.ftl:47]: do we have/are we gonna add force update?  🔨

File: kit/src/components/modal/mod.rs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                                cx.props.on_dismiss.call(());
+                            },
+                            icon: Shape::XMark
+                        },
+                    },
+                    div {
+                        class: "model-body",
+                        rsx!(cx.props.children.as_ref()),
+                },
+
```
Comment (stavares843) [inline kit/src/components/modal/mod.rs:38]: do we need these empty lines? 🔨

File: ui/src/main.rs [hunk 5]
```diff
@@ -966,6 +974,70 @@ fn get_update_icon(cx: Scope) -> Element {
...
+                "*We are going to streamline this process in a future update."
+            },
+            // dl.as_ref().clone().map(|dest| rsx!(
+            //     Button {
+            //         text: "download installer".into(),
+            //         onpress: move |_| {
+            //            on_submit.call(dest.clone());
+            //         }
+            //     }
+            // ))
```
Comment (stavares843) [inline ui/src/main.rs:1035]: do we need this commented out code? 🔨

**Comment by stavares843** [comment]
how can we force the modal to appear? 🔨

**Comment by sdwoodbury** [comment]
this could use a translation
image

File: common/locales/en-US/main.ftl (hunk 3 above)
Comment (WanderingHogan) [reply common/locales/en-US/main.ftl:47]: not sure
for now all manual

File: ui/src/main.rs (hunk 5 above)
Comment (WanderingHogan) [reply ui/src/main.rs:1035]: we do, I didnt want to rip it out, it just wasn't working right now and I didnt have time to look into why

File: kit/src/components/modal/mod.rs (hunk 4 above)
Comment (WanderingHogan) [reply kit/src/components/modal/mod.rs:38]: no

**Pull request merged**